presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -322,8 +325,17 @@ private static boolean checkSchemaMatch(org.apache.parquet.schema.Type parquetTy
             switch (prestoType) {
                 case ROW:
                     if (groupType.getFields().size() == type.getTypeParameters().size()) {
+                        checkArgument(type instanceof RowType, "It must be a RowType here.");
+                        RowType rowType = (RowType) type;
+                        Map<String, Type> prestoFieldMap = rowType.getFields().stream().collect(
+                                Collectors.toMap(
+                                        field -> field.getName().get().toLowerCase(Locale.ENGLISH),
+                                        field -> field.getType()));
                         for (int i = 0; i < groupType.getFields().size(); i++) {
-                            if (!checkSchemaMatch(groupType.getFields().get(i), type.getTypeParameters().get(i))) {
+                            org.apache.parquet.schema.Type parquetFieldType = groupType.getFields().get(i);
+                            String fieldName = parquetFieldType.getName().toLowerCase(Locale.ENGLISH);
+                            Type prestoFieldType = prestoFieldMap.get(fieldName);
+                            if (prestoFieldType == null || !checkSchemaMatch(parquetFieldType, prestoFieldType)) {
```

Comment:
do we need to error out when prestoFieldType == null? If parquet file has a field not exist in Presto metastore, it is OK we just leave it there. What do you think?

Reply:
change the condition to prestoFieldType != null && !checkSchemaMatch(parquetFieldType, prestoFieldType)